src/control/cmd/daos_agent/security_rpc.go

```diff
@@ -146,77 +188,49 @@ func newCachedCredential(key string, cred *auth.Credential, lifetime time.Durati
 	}, nil
 }
 
-// GetMethod gets the corresponding Method for a method ID.
-func (m *SecurityModule) GetMethod(id int32) (drpc.Method, error) {
-	if id == daos.MethodRequestCredentials.ID() {
-		return daos.MethodRequestCredentials, nil
-	}
-
-	return nil, fmt.Errorf("invalid method ID %d for module %s", id, m.String())
-}
-
-func (m *SecurityModule) String() string {
-	return "agent_security"
-}
-
 // HandleCall is the handler for calls to the SecurityModule
-func (m *SecurityModule) HandleCall(ctx context.Context, session *drpc.Session, method drpc.Method, body []byte) ([]byte, error) {
-	if method != daos.MethodRequestCredentials {
-		return nil, drpc.UnknownMethodFailure()
-	}
-
-	return m.getCredential(ctx, session)
-}
-
-// getCredentials generates a signed user credential based on the data attached to
-// the Unix Domain Socket.
-func (m *SecurityModule) getCredential(ctx context.Context, session *drpc.Session) ([]byte, error) {
-	if session == nil {
-		return nil, drpc.NewFailureWithMessage("session is nil")
+func (m *SecurityModule) HandleCall(ctx context.Context, session *drpc.Session, method drpc.Method, reqb []byte) ([]byte, error) {
```

Comment:
The additions in this function seem like they really belong in the GetCredential logic, rather than in the overall SecurityModule.